language/move-bytecode-verifier/src/absint.rs

```diff
@@ -2,17 +2,19 @@
 // Copyright (c) The Move Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+use crate::meter::Meter;
 use move_binary_format::{
     binary_views::FunctionView,
     control_flow_graph::{BlockId, ControlFlowGraph},
+    errors::PartialVMResult,
     file_format::{Bytecode, CodeOffset},
 };
 use std::collections::BTreeMap;
 
 /// Trait for finite-height abstract domains. Infinite height domains would require a more complex
 /// trait with widening and a partial order.
 pub trait AbstractDomain: Clone + Sized {
-    fn join(&mut self, other: &Self) -> JoinResult;
+    fn join(&mut self, other: &Self, meter: &mut impl Meter) -> PartialVMResult<JoinResult>;
```

Comment:
Similar to my comment for type_safety, do we need the meter as an extra argument here? 
Could we just make the `meter` a field on the AbstractDomain implementer? Though that definitely feels a bit funny

Reply:
I totally would have preferred this. Ideally there would be a mut Context going into all this verifier code.